subprojects/core/src/main/java/org/gradle/execution/commandline/CommandLineTaskConfigurer.java

```diff
@@ -94,4 +102,8 @@ private List<String> configureTasksNow(Collection<Task> tasks, List<String> argu
         }
         return remainingArguments;
     }
+
+    private static void allowOnlyOneOption(OptionDescriptor optionDescriptor, CommandLineParser parser) {
+        parser.allowOneOf(optionDescriptor.getName(), optionDescriptor.getName().substring(3));
```

Comment:
:art: I think this should be reworked. The knowledge of doing substrings is spread apart, this should be in a central place or encapsulated.